common/lib/client/realtimechannel.js

```diff
@@ -409,68 +423,47 @@ var RealtimeChannel = (function() {
 		}
 		this.setInProgress(syncOp, syncInProgress);
 		this.presence.setAttached();
-		this.setState('attached');
+		this.notifyState('attached');
 	};
 
-	RealtimeChannel.prototype.setDetached = function(message) {
-		this.clearStateTimer();
+	RealtimeChannel.prototype.notifyState = function(state, reason) {
+		Logger.logAction(Logger.LOG_MICRO, 'RealtimeChannel.notifyState', 'name = ' + this.name + ', current state = ' + this.state + ', notifying state ' + state);
+    this.clearStateTimer();
 
-		var msgErr = message.error;
-		if(msgErr) {
-			var err = ErrorInfo.fromValues(message.error);
-			this.setState('detached', err);
-			this.failPendingMessages(err);
-		} else {
-			/* Don't bother with setState if there's no err and no statechange */
-			if(this.state !== 'detached') {
-				this.setState('detached');
-			}
-			this.failPendingMessages(new ErrorInfo('Channel detached', 90001, 404));
+		if(state === this.state) {
+			return;
 		}
-	};
-
-	RealtimeChannel.prototype.setFailed = function(message) {
-		this.clearStateTimer();
-		var err = ErrorInfo.fromValues(message.error || {statusCode: 400, code: 90000, message: 'Channel failed'});
-		this.setState('failed', err);
-		this.failPendingMessages(err);
-	};
-
-	RealtimeChannel.prototype.setSuspended = function(err, suppressEvent) {
-		if(this.state !== 'detached' && this.state !== 'failed') {
-			Logger.logAction(Logger.LOG_MINOR, 'RealtimeChannel.setSuspended', 'deactivating channel; name = ' + this.name + ', err ' + (err ? err.message : 'none'));
-			this.clearStateTimer();
-			this.presence.setSuspended(err);
-			if(!suppressEvent) {
-				this.setState('detached');
-			}
-			this.failPendingMessages(err);
+    if(state !== 'attached' && state !== 'attaching') {
+      this.presence.setSuspended(reason || RealtimeChannel.invalidStateError);
+      this.failPendingMessages(reason || RealtimeChannel.invalidStateError);
+    }
+		if(state === 'suspended' && this.connectionManager.state.sendEvents) {
```

Comment:
whether you can send events in that state. (i.e. basically just `connected`). Leftover from the days when 'messages' were called 'events' (with a `queueEvents` client option, etc)